ibm/service/power/resource_ibm_pi_capture.go

```diff
@@ -198,11 +198,13 @@ func resourceIBMPICaptureCreate(ctx context.Context, d *schema.ResourceData, met
 		if err != nil {
 			log.Printf("Error on get of ibm pi capture (%s) while applying pi_user_tags: %s", capturename, err)
 		}
-		if imagedata.Crn != "" {
-			oldList, newList := d.GetChange(Arg_UserTags)
-			err = flex.UpdateGlobalTagsUsingCRN(oldList, newList, meta, string(imagedata.Crn), "", UserTagType)
-			if err != nil {
-				log.Printf("Error on update of pi capture (%s) pi_user_tags during creation: %s", *imagedata.ImageID, err)
+		if imagedata != nil {
```

Comment:
Don't we want to return the error in the above err condition instead of an additional check here?

Reply:
Because this is part of updating the tags on the global tagging service, we decided we do not want to throw an error since the resource has been created

Reply:
But it concerns me that new image does not exist could mean capture did not complete successfully. 
It could be that since you are querying by name the cache on getimage api is not ready for new image. Could you check with SB why the name query fails to fetch the image details?

Reply:
I believe there is a way to check the error type but just to make sure I am understanding, you want to throw an error during create in the case of a "Not found" error, correct?

Reply:
Update: When originally testing this, the error was never hit because an image was returned correctly every time by SB at the time. I was hitting this issue today when testing it due to an error in service broker resolving the name to the incorrect ID.

Reply:
Update: When originally testing this, the error was never hit because an image was returned correctly every time by SB at the time. I was hitting this issue today when testing it due to an error in service broker. Normally this did not happen.

Reply:
I want to ensure that the image by name works as soon as the new image is created. This condition check should not hide that fact and always work without updating the tag.